addons/website/static/src/snippets/s_table_of_content/options.js

```diff
@@ -50,9 +71,29 @@ options.registry.TableOfContent = options.Class.extend({
         const $nav = this.$target.find('.s_table_of_content_navbar');
         const $headings = this.$target.find(this.targetedElements);
         $nav.empty();
+        const uniqueHeadingIds = new Set();
         _.each($headings, el => {
             const $el = $(el);
-            const id = 'table_of_content_heading_' + _.now() + '_' + _.uniqueId();
+            if (!this.$target[0].dataset.tocId) {
+                const tocEls = this.$target[0].closest("#wrapwrap").querySelectorAll("section.s_table_of_content");
+                const tocIds = [...tocEls].map(tocEl => parseInt(tocEl.dataset.tocId) || 0);
+                const nextTocId = 1 + Math.max(...tocIds);
+                this.$target[0].dataset.tocId = nextTocId;
+            }
```

Comment:
I don't think you should do this in the `_.each`, surely you cannot lose the `this.$target[0].dataset.tocId` at each iteration?